File: Backend/rest/src/main/java/net/chrisparton/sparkled/rest/EasingTypeRestService.java
package net.chrisparton.sparkled.rest;

import net.chrisparton.sparkled.model.animation.easing.EasingTypeCode;
import net.chrisparton.sparkled.model.animation.easing.reference.EasingTypes;

import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

@Path("/easing-type")
public class EasingTypeRestService extends RestService {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    public Response get() {
        return getJsonResponse(EasingTypes.get());
    }
}

File: rest/src/main/kotlin/io/sparkled/rest/service/easingtype/EasingTypeRestService.kt
package io.sparkled.rest.service.easingtype

import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/easingTypes")
class EasingTypeRestService
@Inject constructor(private val handler: EasingTypeRestServiceHandler) {

    @GET
    @Produces(MediaType.APPLICATION_JSON)
    fun get(): Response {
        return handler.get()
    }
}